app/api/managers.py

```diff
@@ -31,3 +31,18 @@ def get_label_per_data(self, project):
             user_count[d['user__username']] += d['user__count']
 
         return label_count, user_count
+
+
+class QandAAnnotationManager(Manager):
```

Comment:
The three annotation managers are now very similar. I wonder if it'll soon be worth refactoring them... just some thoughts to get a conversation going:

```py
class BaseAnnotationManager(Manager):
    pivots = ()
    aggregates = ()

    def get_label_per_data(self, project):
        label_count = Counter()
        user_count = Counter()
        docs = project.documents.all()
        annotations = self.filter(document_id__in=docs.all())

        pivot1, pivot2 = self.pivots
        agg1, agg2 = self.aggregates
        for d in annotations.values(pivot1, pivot2).annotate(Count(agg1), Count(agg2)):
            label_count[d[pivot1]] += d[f'{agg1}__count']
            user_count[d[pivot2]] += d[f'{agg2}__count']

        return label_count, user_count


class AnnotationManager(BaseAnnotationManager):
    pivots = ('label__text', 'user__username')
    aggregates = ('label', 'user')


class Seq2seqAnnotationManager(BaseAnnotationManager):
    pivots = ('text', 'user__username')
    aggregates = ('text', 'user')


class QandAAnnotationManager(BaseAnnotationManager):
    pivots = ('response', 'user__username')
    aggregates = ('response', 'user')
```

Reply:
I definitly agree that some factorisation could happen soon.